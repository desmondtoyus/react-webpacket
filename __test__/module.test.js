'use strict'
const modules = require('./../contents/module');

describe('viewTemplate()', function () {
    it('Should exist', function () {
        expect(modules.viewTemplate).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.viewTemplate).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.viewTemplate('home');
        expect(result).toMatchSnapshot();
    });
    it('should return error', function () {
        expect(modules.viewTemplate()).toBeFalsy()
    });

});

describe('reducerTemplate()', function () {
    it('Should exist', function () {
        expect(modules.reducerTemplate).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.reducerTemplate).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.reducerTemplate('shared');
        expect(result).toMatchSnapshot();
    });
    it('should return error', function () {
        expect(modules.reducerTemplate()).toBeFalsy();
    });
    
});
describe('actionTemplate()', function () {
    it('Should exist', function () {
        expect(modules.actionTemplate).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.actionTemplate).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.actionTemplate('shared');
        expect(result).toMatchSnapshot();
    });
    it('should return error', function () {
        expect(modules.actionTemplate()).toBeFalsy();
    });
});
describe('pageUrls()', function () {
    it('Should exist', function () {
        expect(modules.pageUrls).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.pageUrls).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.pageUrls(['home']);
        expect(result).toMatchSnapshot();
    });
    it('should return false for string', function () {
        const result = modules.pageUrls('home');
        expect(result).toBeFalsy();
    });
    it('should return error', function () {
        expect(modules.pageUrls()).toBeFalsy();
    });
});
describe('rootReducers()', function () {
    it('Should exist', function () {
        expect(modules.rootReducers).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.rootReducers).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.rootReducers(['shared']);
        expect(result).toMatchSnapshot();
    });
    it('should return false for string', function () {
        const result = modules.rootReducers('shared');
        expect(result).toBeFalsy();
    });
    it('should return error', function () {
        expect(modules.rootReducers()).toBeFalsy();
    });
});

describe('configStore()', function () {
    it('Should exist', function () {
        expect(modules.configStore).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.configStore).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.configStore(['shared']);
        expect(result).toMatchSnapshot();
    });
    it('should return false for string', function () {
        const result = modules.configStore('shared');
        expect(result).toBeFalsy();
    });
    it('should return error', function () {
        expect(modules.configStore()).toBeFalsy();
    });
});

describe('caddActionToType()', function () {
    it('Should exist', function () {
        expect(modules.addActionToType).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.addActionToType).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.addActionToType(['ADD_USER']);
        expect(result).toMatchSnapshot();
    });
    it('should return false for string', function () {
        const result = modules.addActionToType('ADD_USER');
        expect(result).toBeFalsy();
    });
    it('should return error', function () {
        expect(modules.addActionToType()).toBeFalsy();
    });
});

describe('createApp()', function () {
    it('Should exist', function () {
        expect(modules.createApp).toBeDefined();
    });
    it('should be a function', function () {
        expect(modules.createApp).toBeInstanceOf(Function);
    });
    it('should return a string', function () {
        const result = modules.createApp(['home']);
        expect(result).toMatchSnapshot();
    });
    it('should return false for string', function () {
        const result = modules.createApp('home');
        expect(result).toBeFalsy();
    });
    it('should return error', function () {
        expect(modules.createApp()).toBeFalsy();
    });
});

// createApp

