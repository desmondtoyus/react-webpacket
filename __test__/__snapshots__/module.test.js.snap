// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actionTemplate() should return a string 1`] = `
"import { SHARED } from './types';
    export const sharedInputChange = ({ prop, value }) => (dispatch) => {
    dispatch({ type: SHARED, prop, value });
};"
`;

exports[`caddActionToType() should return a string 1`] = `
"export const INPUT_CHANGE = 'INPUT_CHANGE';
 export const FETCH_DATA = 'FETCH_DATA';
export const  ADD_USER = 'ADD_USER'; 
"
`;

exports[`configStore() should return a string 1`] = `
"import { combineReducers } from 'redux';
import {
    sampleReducer,
  sharedReducer, 

} from './reducers';

const configureStore = combineReducers({
sample: sampleReducer,
shared : sharedReducer, 

});
export default configureStore;"
`;

exports[`createApp() should return a string 1`] = `
"import React from 'react';
import { Helmet } from 'react-helmet';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import Footer from './components/Footer';
import Header from './components/Header';
import SamplePage from './views/SamplePage';
import Home from './views/home'; 

import PageNotFound from './views/PageNotFound';

const App = () => (
    <div className=\\"App\\">
        <Helmet titleTemplate=\\"%s - SiteName\\" defaultTitle=\\"Some default title\\">
            <meta name=\\"description\\" content=\\"some description content\\"/>
        </Helmet>
        {/* BrowserRouter comes with default history */}
        <BrowserRouter>
            <div>
                <Header/>
                <div className=\\"main\\">
                    <Switch>
                        <Route exact path=\\"/\\" component={SamplePage}/>
                                    <Route exact path=\\"/home\\" component={Home} /> 

                        <Route path=\\"\\" component={PageNotFound}/>
                    </Switch>
                    <Footer/>
                </div>
            </div>
        </BrowserRouter>
    </div>
);

export default App;"
`;

exports[`pageUrls() should return a string 1`] = `
"
  import React from 'react';
  import { Link } from 'react-router-dom';
  const Views = () => (
    <div className=\\"col-md-6\\">
            <div className=\\"panel panel-success\\">
                <div className=\\"panel-heading\\">
                    <h3 className=\\"panel-title\\">
                        My Pages</h3>
                </div>
                <ul className=\\"list-group\\">
                <Link to=\\"/\\" className=\\"list-group-item active\\"> Sample Page </Link>
    <Link className=\\"list-group-item active\\" to=\\"/home\\"> Home </Link>
    </ul>
    </div>
</div>
  );
  export default Views;"
`;

exports[`reducerTemplate() should return a string 1`] = `
"import { SHARED } from '../actions/types';

    const INITIAL_STATE = {
      name: '',
    };
    
    export default function (state = INITIAL_STATE, action) {
      switch (action.type) {
        case SHARED:
          return {
            ...state,
            [action.prop]: action.value,
          };
        default:
          return state;
      }
    };"
`;

exports[`rootReducers() should return a string 1`] = `
"import sampleReducer from './samplepage.reducer';  
import sharedReducer from './shared.reducer'; 

export {  sampleReducer, 
 sharedReducer, 
};"
`;

exports[`viewTemplate() should return a string 1`] = `
"import React from 'react';
    import { Helmet } from 'react-helmet';
    
    class Home extends React.Component {
      constructor(props) {
        super(props);
        this.state = {
    
        };
      }
    
      render() {
        return (
          <div>
            <Helmet>
              <title>Home </title>
              <meta name=\\"description\\" content=\\"Home description\\" />
            </Helmet>
            <h1>HOME Page!</h1>
    
          </div>
        );
      }
    }
    
    export default Home;"
`;
